homeassistant/components/mqtt/subscription.py

```diff
@@ -0,0 +1,51 @@
+"""
+Helper to handle a set of topics to subscribe to.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/mqtt/
+"""
+import logging
+
+from homeassistant.components import mqtt
+from homeassistant.components.mqtt import DEFAULT_QOS, MessageCallbackType
+from homeassistant.loader import bind_hass
+from homeassistant.helpers.typing import (
+    HomeAssistantType)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@bind_hass
+async def async_subscribe_topics(hass: HomeAssistantType, sub_state: dict,
+                                 new_topics: dict,
+                                 msg_callback: MessageCallbackType,
```

Comment:
We should not have a generic msg callback for all topics. For example, this won't work for the MQTT light, as it has [a different callback](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/light/mqtt.py#L276-L297) for brightness and state.

I would expect `new_topics` to be called `topics` (as there might nothing be new about it) and be defined like this:

```python
{
  '/home/light/kitchen/state': {
    'qos': 2,
    'encoding': 'utf-8',
    'msg_callback': state_received
  }
}
```